pkg/configmap/configmap.go

```diff
@@ -42,12 +41,16 @@ func ValidateOperatorConfigMap(configmap OperatorConfigMap) error {
 func GetOperatorConfigMap(kubeClient client.Client) (OperatorConfigMap, error) {
 	var OperatorConfigMap OperatorConfigMap
 	configmap := &corev1.ConfigMap{}
-	if err := kubeClient.Get(context.TODO(), kubetypes.NamespacedName{Name: OperatorConfigMapName, Namespace: operatorConfigMapNamespace}, configmap); err != nil {
+	if err := kubeClient.Get(context.TODO(), kubetypes.NamespacedName{Name: OperatorConfigMapName, Namespace: OperatorConfigMapNamespace}, configmap); err != nil {
 		return OperatorConfigMap, fmt.Errorf("unable to get configmap: %v", err)
 	}
 
-	if err := mapstructure.Decode(configmap.Data, &OperatorConfigMap); err != nil {
-		return OperatorConfigMap, fmt.Errorf("unable to unmarshal configmap: %v", err)
+	if data, ok := configmap.Data[OperatorConfigMapKey]; !ok {
```

Comment:
this will try to get `configmap.Data["data"]` right? Is this correct?

Reply:
Yes, it requires the `data` key.

Reply:
Can't we find a way to parse the existing format? I like the parsing, but having
```
data:
  data:
```
seems unclean to me.

Reply:
It is hard to still parse the existing format. Since config map itself is `map[string]string`, if we want to have a list, then we only have 2 choices:

1. Have a single key, and put all our configs to the value. If you think the key name `data` not good, I can update it to something else, like `config.yaml`, so people will know it is actually a yaml config file. WDYT?

``` yaml
data:
  config.yaml: |
    billingAccount: xxx
    folderID: xxx
    consoleAccess: [item1, item2]
```

2. Have multiple keys. In this case, we have to find a way to make consoleAccess value a string. TBH it might doable, but maybe mess up things.

``` yaml
data:
  billingAccount: xxx
  folderID: xxx
  consoleAccess: "item1, item2"
```

Reply:
I am still looking at this and I find a way to get this working for both formats.

``` yaml
data:
  billingAccount: xxx
  folderID: xxx
  config.yaml: |
    billingAccount: xxx
    folderID: xxx
    consoleAccess: [item1, item2]
```

Reply:
yes @yeya24 let's do it like this and switch to the new format now, I think it's the cleaner way of defining and parsing a configmap. Some thoughts:
- Make sure it works if only the "old" fields are set
- Don't call it `data`, maybe `config.yaml` would be a good alternative.
- Let's not defer cleaning up the redundant fields but prepare a PR right away so we can clean it up as soon as all environments are migrated.

Reply:
👍 to `config.yaml` key name.